raft/node.go

```diff
@@ -381,13 +395,17 @@ func (n *node) run(r *raft) {
 			if !IsEmptySnap(rd.Snapshot) {
 				prevSnapi = rd.Snapshot.Metadata.Index
 			}
+			if index := rd.appliedCursor(); index != 0 {
```

Comment:
rd.CanApplyTo? The index returned here is the index the application can apply to rather than what the application has applied.